ComponentKit/Core/Scope/CKComponentScopeRootFactory.h

```diff
@@ -0,0 +1,17 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+
+@protocol CKComponentStateListener;
+
+@class CKComponentScopeRoot;
+
+CKComponentScopeRoot *CKComponentScopeRootWithListener(id<CKComponentStateListener> listener);
```

Comment:
Hm. I'm not sure we would necessarily want to override the predicates but instead rely on a separate set of predicates. Right? There shouldn't be a need to replace the appearance event predicates when incorporating additional working range predicates.

I'm having a bit of trouble seeing how we could leverage this new scope API outside where it's being used internally by ComponentKit. Let's sync up on this when you have some time.